api/migrations/0003_fake_data.py

```diff
@@ -0,0 +1,329 @@
+            start_date=self.faker.past_date,
+            end_date=self.faker.future_date,
+            **self._get_group_days_of_week(),
+        )
+
+    def _make_speaking_club_recipe(self) -> Recipe:
+        group_common_recipe = self._make_group_common_recipe(APP_NAME + ".SpeakingClub")
+        return group_common_recipe.extend(
+            is_for_children=self.faker.pybool,
+            language=lambda: self.faker.random_element(Language.objects.all()),
+            teachers_under_18=lambda: related(
+                *[self.teacher_under_18]
+                * self.faker.pyint(
+                    min_value=MIN_AMOUNT_OF_TEACHERS_UNDER_18_IN_SPEAKING_CLUB,
+                    max_value=MAX_AMOUNT_OF_TEACHERS_UNDER_18_IN_SPEAKING_CLUB,
+                )
+            ),
+        )
+
+
+class FakeDataPopulator(DataPopulator):
+    def __init__(self, apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+        super().__init__(apps, schema_editor)
+        self.recipes = RecipeStorage()
+
+    def _make_amount_of_recipe_and_skip_database_error(
+        self, recipe: Recipe, amount: int
+    ) -> None:
+        """
+        Ensures that given amount of objects is made.
+
+        While generating objects by recipe, it is possible that some of them will violate any db constraint.
+        We need to recreate such objects to skip db constraints and be sure that needed amount is generated.
+        """
+        for _ in range(amount):
+            while True:
+                try:
+                    recipe.make()
+                    break
+                except DatabaseError:
+                    pass
+
+    def _make_fake_coordinators_without_group(self):
+        """Makes fake coordinators without group."""
+        self.recipes.coordinator.make(_quantity=AMOUNT_OF_COORDINATORS_WITHOUT_GROUP)
+
+    def _make_fake_students_without_group(self):
+        """Makes fake students without group."""
+        self.recipes.student.make(_quantity=AMOUNT_OF_STUDENTS_WITHOUT_GROUP)
+
+    def _make_fake_teachers_without_group(self):
+        """Makes fake teachers without groups."""
+        self.recipes.teacher.make(_quantity=AMOUNT_OF_TEACHERS_WITHOUT_GROUP)
+
+    def _make_fake_teachers_under_18_without_speaking_club(self):
+        """Makes fake teachers under 18 without_speaking_club."""
+        self.recipes.teacher_under_18.make(
+            _quantity=AMOUNT_OF_TEACHERS_UNDER_18_WITHOUT_SPEAKING_CLUB
+        )
+
+    def _make_fake_groups(self):
+        """Makes fake groups."""
+        self._make_amount_of_recipe_and_skip_database_error(
+            self.recipes.group, AMOUNT_OF_GROUPS
+        )
+
+    def _make_fake_speaking_clubs(self):
+        """Makes fake speaking_clubs."""
+        self._make_amount_of_recipe_and_skip_database_error(
+            self.recipes.speaking_club, AMOUNT_OF_SPEAKING_CLUBS
+        )
+
+    def _populate(self):
+        """Runs pre-population operations."""
```

Comment:
```suggestion
        """Runs operations required for populating the database with fake data."""
```

Can look a bit strange that `_populate` does "pre-population".